internal/server/data/data.go

```diff
@@ -22,22 +22,33 @@ import (
 )
 
 // NewDB creates a new database connection and runs any required database migrations
-// before returning the connection.
-func NewDB(connection gorm.Dialector) (*gorm.DB, error) {
-	db, err := NewRawDB(connection)
+// before returning the connection. The loadDBKey function is called after
+// initializing the schema, but before any migrations.
+func NewDB(connection gorm.Dialector, loadDBKey func(db *gorm.DB) error) (*gorm.DB, error) {
```

Comment:
This is the primary change. Instead of having tests use `NewDB`, and production code using `NewRawDB` this allows both tests and production to use the same `NewDB` function.

It would be nice to have tests also use the same `loadDBKey` function, but that is a larger refactor, so I left it for another time. 

This change allows us to unexport `migrate` and `preMigrate`, so that we expose a smaller interface to other packages.